routes/authors.js

```diff
@@ -1,8 +1,82 @@
 const express = require("express");
 const generateId = require("uuid/v4");
+const authors = require('../data/authors.json')
+const books = require('../data/books.json')
 
 const router = express.Router();
 
 // Define your endpoints here
 
-module.exports = router;
+router.get('/', (req, res, next) => {
+  res.json({
+    authors
+  })
+})
+
+router.post('/', (req, res, next) => {
+  const newAuthor = {
+    id: generateId(),
+    name: req.body.name,
+    website: req.body.website
+  };
+  authors.push(newAuthor)
+  res.json({
+    newAuthor,
+    authors
+  })
+})
+
+router.get('/:id', (req, res, next) => {
+  const id = req.params.id
+  authors.forEach(author => {
+    if (id === author.id) {
+      res.json({
+        author
+      })
+    }
+  })
+  res.json({
+    error: "Error looking up author"
```

Comment:
Props on you for handling the error case 💯 .

However, there are two possible improvements:

- Using `res.json` doesn't stop the function, meaning that you are sending something twice. Use a conditional or a return earlier to make sure you only send something once.
- You should use a special HTTP code to make it an error; it this case error 404 is appropriate.